packages/react-cookie/src/CookieUniversalProvider.tsx

```diff
@@ -0,0 +1,26 @@
+import React from 'react';
+import {useNetworkManager} from '@shopify/react-network';
+import {useLazyRef} from '@shopify/react-hooks';
+import {CookieManager} from './manager';
+import {CookieContext} from './context';
+import {hasDocumentCookie} from './utilities';
+
+export interface Props {
+  children?: React.ReactNode;
+}
+
+export function CookieUniversalProvider({children}: Props) {
+  const manager = useNetworkManager();
+
+  const cookieManager = useLazyRef(() => {
+    return !hasDocumentCookie() && manager
```

Comment:
this seems slightly wrong to me since it could make a front-end focused cookie manager if the user forgot to wrap their tree in a network provider or it isn't there for some other weird reason. I'd rather we do something like

```tsx
if (hasDocumentCookie()) {
  return new CookieManager();
}

if (manager == null) {
  throw new error('You fool! You fell victim to one of the classic blunders! Never server render a react-html app without a network provider when death is on the line!');
}

return manager.cookies;
```